lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -2994,13 +2908,13 @@ def _get_os_disk_size(self, azure_runbook: AzureNodeSchema) -> int:
                 # if no image info, use default size 30
                 return 30
         elif azure_runbook.shared_gallery:
-            azure_runbook.shared_gallery = self._parse_shared_gallery_image(
-                azure_runbook.shared_gallery
+            azure_runbook.shared_gallery = azure_runbook.shared_gallery._parse_version(
```

Comment:
Can it be called automatically, so it doesn't need to be called explicitly here.

Reply:
@squirrelsc
I could move all the 'os disk size' logic to be in the AzureImageSchema's.
This will add to the size/complexity of the PR, but I can do that if you think it is worth it.

Reply:
You can do it in a commit after the main change.